src/python/pants/backend/python/subsystems/python_native_code.py

```diff
@@ -0,0 +1,123 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from collections import defaultdict
+
+from pants.backend.native.subsystems.native_toolchain import NativeToolchain
+from pants.backend.native.targets.native_library import NativeLibrary
+from pants.backend.python.subsystems.python_setup import PythonSetup
+from pants.backend.python.targets.python_binary import PythonBinary
+from pants.backend.python.targets.python_distribution import PythonDistribution
+from pants.base.exceptions import IncompatiblePlatformsError
+from pants.subsystem.subsystem import Subsystem
+from pants.util.memo import memoized_property
+from pants.util.objects import Exactly
+
+
+class PythonNativeCode(Subsystem):
+  """A subsystem which exposes components of the native backend to the python backend."""
+
+  options_scope = 'python-native-code'
+
+  default_native_source_extensions = ['.c', '.cpp', '.cc']
```

Comment:
It's a bit concerning that this is redundant with the `cpp-compile` and `c-compile` subsystems... are different tools being invoked that do not all use the same extensions? Would it be reasonable to just default to supporting the intersection of these two sets, and then configure it on the relevant language subsystem?

Reply:
This shouldn't be here at all, but checking if a `python_dist()` target has "native" sources has been done by checking whether it has sources by any of these extensions. What it should really be doing is scanning the requirements closure to see if any of them have platform-specific code. That somehow didn't get made into a followup issue, which I will do now.

Reply:
Made into #5949.